### PR TITLE
All packages to PackageReference

### PR DESCRIPTION
**Pull request by agr**
Current master branch builds produce broken packages: for example, `NuGet.Services.Messaging.Email.nupkg` includes `NuGet.Services.Entities` in the package instead of referenced as a dependency. This effectively prevents entity updates in Gallery (`Add-Migration` produces empty migrations), if one only updates `NuGet.Services.Entities` reference without updating the `NuGet.Services.Messaging.Email` reference.
This happens due to some issues between PC, project.json and PR projects.
This PR 
* makes all projects to use PackageReference;
* Updates build script to use `New-ProjectPackage` to produce .nupkgs;
* Drops the NupkgWrench call from the build script.

### CODE REVIEW

**Comment by joelverhagen** [comment]
Does this supersede https://github.com/NuGet/ServerCommon/pull/214/files?

**Comment by agr** [comment]
@joelverhagen, yes

**Comment by chenriksson** [comment]
@agr @joelverhagen Since PackageRef is a big change affecting dependencies, can we also apply other standardizations at the same time?
- Target Net462
- Tools version "15"
- Autogenerate binding redirects
- Dependency cleanup (remove transitive dependencies, which should no longer be necessary)

Cleanup of dependencies would help us get to the point of flipping the WarnAsError switch.

**Comment by agr** [comment]
There is only one project that generates executable: `AzureSqlConnectionTest` and for it auto generate binding redirects is already enabled. As far as I understand, this option makes no sense for the libraries.

Regarding the transitive dependencies: most of the projects updated (all except one) were project.json projects, for which transitive dependencies were not supposed to be listed in the first place.

Tools version and target framework were updated.

**Comment by agr** [comment]
I looked through the nuspecs and contents of the packages produced to make sure no extra files are included and proper dependencies are listed. But ultimately, the indication of correctness is that services that use it does not break, so we'll have to update and deploy.

File: src/NuGet.Services.Build/NuGet.Services.Build.csproj [hunk 1]
```diff
@@ -52,20 +56,23 @@
...
+      <Version>0.3.0</Version>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="Microsoft.VisualStudio.Setup.Configuration.Interop">
+      <Version>1.8.24</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Build.Tasks.Pack">
+      <Version>4.8.0</Version>
+      <PrivateAssets>all</PrivateAssets>
```
Comment (joelverhagen) [inline src/NuGet.Services.Build/NuGet.Services.Build.csproj]: nit, this should have
```xml
      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
```

4.8.0 is not marked as a development dependency, but it should be treated as one. Therefore, it should have the same asset rules as `MicroBuild.Core`.

Comment (agr) [reply src/NuGet.Services.Build/NuGet.Services.Build.csproj]: This was taken from [NuGetGallery.Core.csproj](https://github.com/NuGet/NuGetGallery/blob/9070951769c2a57f7d56746bdec48938df80bf94/src/NuGetGallery.Core/NuGetGallery.Core.csproj#L292-L295). Do we need to update it there, too?

**Pull request merged**